DEVDOCS-3855: [net new] Data Solutions API (#1491)
bigcommerce/dev-docs: docs/api-docs/provider-apis/data-solutions/data-solutions-overview.md

# Data Solutions API

> The Data Solutions API allows you to configure **storefront** channel settings for a store's prebuilt data analytic solutions.  

Reviewer: ```suggestion
The Data Solutions API lets you configure **storefront** channel settings for a store's prebuilt data analytic solutions.  
```
https://developers.google.com/style/word-list#allows-you-to

> Note that a store has global settings for data solutions, from which any storefront channel can inherit. A merchant can override these global settings with storefront-specific settings for their data solutions. This API configures settings for **storefront** channels at the store's global level (`channel_id` is 0) and at the specific storefront channel level.   

Reviewer: ```suggestion
Note that a store has global settings for data solutions, from which any storefront channel can inherit. A merchant can override these global settings with storefront-specific settings for their data solutions. This API configures settings for **storefront** channels at the store's global level (`channel_id` is `0`) and at the specific storefront channel level.   
```

> This article introduces how to configure web analytics using the Data Solutions API. Because each web analytic requires different fields in the request and/or response, this article is divided by web analytic.

Reviewer 2: ```suggestion
This article introduces how to configure web analytics using the Data Solutions API. Because each web analytic requires different fields in the request and/or response, this article is divided by web analytics.
```

> You can get **all** web analytics, get a **single** web analytic, or **update** a single analytic. To get or update a **single** web analytic, the `id` of the Web Analytic must be specified in the path.

Reviewer 2: ```suggestion
You can get **all** web analytics, get a **single** web analytic, or **update** a single analytic. To get or update a **single** web analytic, the `id` of the web analytic must be specified in the path.
```

> | Web Analytic | Id |

Reviewer 2: ```suggestion
| Web analytic | ID |
```

| ------------- | -------- |
| Google Analytics | 1 |
| Visual Website Optimizer | 2 |
| Facebook Pixel | 3 |
| Segment | 4 |
| Site Verification Tags | 6 |
| Affiliate Conversion Tracking | 7 |

> **Note**: Web Analytic id 5 is no longer in use.

Reviewer 2: ```suggestion
**Note**: Web analytic ID 5 is no longer in use.
```

> ## Google Analytics

Reviewer 2: ```suggestion
## Google analytics
```

Reviewer: Google Analytics is a proper noun here, so I think it can be left capitalized

| Field | Type | Description |
| ------------- | -------- |  -------- |

> | id | integer | Id of the Web Analytic. `1` for Google Analytics |

Reviewer 2: ```suggestion
| id | integer | ID of the web analytic. `1` for Google Analytics |
```

> | channel_id | integer | Id of the storefront channel. Default is `0`. |

Reviewer 2: ```suggestion
| channel_id | integer | ID of the storefront channel. Default is `0`. |
```

> | name | string | Name of Web Analytic `Google Analytics` |

Reviewer 2: ```suggestion
| name | string | Name of web analytic `Google Analytics` |
```

> | enabled | boolean | Indicates whether merchant has [enabled Google Analytics](https://support.bigcommerce.com/s/article/Data-Solutions?language=en_US#web-analytics) in their store |

Reviewer 2: ```suggestion
| enabled | boolean | Indicates whether the merchant has [enabled Google Analytics](https://support.bigcommerce.com/s/article/Data-Solutions?language=en_US#web-analytics) in their store |
```

> | version | integer | Corresponds to connection field that merchant is using to connect to Google Analytics. Value is `1` when tracking code is used. Value is `2` when property id is used. |

Reviewer 2: ```suggestion
| version | integer | Corresponds to connection field that merchant is using to connect to Google analytics. Value is `1` when the tracking code is used. Value is `2` when the property id is used. |
```

| data_tag_enabled | boolean | Indicates whether data tags are enabled by Google Analytics |

> | tracking_code | string | Tracking code merchant uses to connect Google Analytics to store. Only returned if version is `1`. |

Reviewer 2: ```suggestion
| tracking_code | string | Tracking code merchant uses to connect Google Analytics to store. Only returned if the version is `1`. |
```

> | property_id | string | Property id merchant uses to connect Google Analytics to store. Only returned if version is `2`.  |

Reviewer 2: ```suggestion
| property_id | string | Property id merchant uses to connect Google Analytics to store. Only returned if the version is `2`.  |
```

> | property_id | string | Property id merchant uses to connect Google Analytics to store. Only returned if version is `2`.  |

Reviewer: ```suggestion
| property_id | string | Property ID merchant uses to connect Google Analytics to store. Only returned if version is `2`.  |
```

> The version corresponds with the "Connect with Field" that a merchant is using to connect to Google Analytics. This connection field affects the fields that are requested and returned in [Get the Google Analytic](#get-the-google-analytic) and [Update the Google Analytic](#update-the-google-analytic).

Reviewer 2: ```suggestion
The version corresponds with the "Connect with Field" that a merchant uses to connect to Google Analytics. This connection field affects the fields that are requested and returned in [Get the Google Analytic](#get-the-google-analytic) and [Update the Google Analytic](#update-the-google-analytic).
```

![Version on Google Analytics](https://storage.googleapis.com/bigcommerce-production-dev-center/images/Version%20for%20Google%20Analytics.png).

### Get the Google Analytic

> When a merchant uses tracking code for the connection field, your response will have a version of `1` as well as the `tracking_code` field. If a merchant has not entered a tracking code, `tracking_code` will return as an empty string.

Reviewer 2: ```suggestion
When a merchant uses a tracking code for the connection field, your response will have a version of `1` and the `tracking_code` field. If a merchant has not entered a tracking code, the `tracking_code` will return as an empty string.
```

```JSON title="Sample Request" lineNumbers
GET https://api.bigcommerce.com/stores/{{STORE_HASH}}/v3/settings/analytics/{id}
X-Auth-Token: {{ACCESS_TOKEN}}
Content-Type: application/json
Accept: application/json
```

```JSON title="Sample Response" lineNumbers
{
    "data": {
        "id": 1,
        "channel_id": 0,
        "name": "Google Analytics",
        "enabled": true,
        "version": 1,
        "data_tag_enabled": true,
        "tracking_code": "GA-1234567890"
    },
    "meta": {}
}
```

> When a merchant uses property id for the connection field, your response will have a version of `2` as well as the `property_id` field. If a merchant has not entered a property ID, `property_id` will return as an empty string.

Reviewer 2: ```suggestion
When a merchant uses property id for the connection field, your response will have a version of `2` and the `property_id` field. If a merchant has not entered a property ID, `property_id` will return as an empty string.
```

X-Auth-Token: {{ACCESS_TOKEN}}
Content-Type: application/json
Accept: application/json
```

```JSON title="Sample Response" lineNumbers
{
    "data": {
        "id": 1,
        "channel_id": 0,
        "name": "Google Analytics",
        "enabled": true,
        "version": 2,
        "data_tag_enabled": true,
        "property_id": "GA-1234567890"
    },
    "meta": {}
}
```
### Update the Google Analytic

```JSON title="Sample Request" lineNumbers
PUT https://api.bigcommerce.com/stores/{{STORE_HASH}}/v3/settings/analytics{id}
X-Auth-Token: {{ACCESS_TOKEN}}
Content-Type: application/json
Accept: application/json

{
    "id": 1,
    "channel_id": 0,
    "name": "Google Analytics",
    "enabled": true,
    "version": 2,
    "data_tag_enabled": true,
    "property_id": "GA-1234567890"
}
```

```JSON title="Sample Response" lineNumbers
{
    "data": {
        "id": 1,
        "channel_id": 0,
        "name": "Google Analytics",
        "enabled": true,
        "version": 2,
        "data_tag_enabled": true,
        "property_id": "GA-1234567890"
    },
    "meta": {}
}
```
&nbsp;

## Visual Website Optimizer

| Field | Type | Description |
| ------------- | -------- |  -------- |

> | id | integer | Id of the Web Analytic. `2` for Visual Website Optimizer |

Reviewer 2: ```suggestion
| id | integer | ID of the Web analytic. `2` for Visual Website Optimizer |
```

> | channel_id | integer | Id of the storefront channel. Default is `0`. |

Reviewer 2: ```suggestion
| channel_id | integer | ID of the storefront channel. Default is `0`. |
```

> | name | string | Name of Web Analytic `Visual Website Optimizer` |

Reviewer 2: ```suggestion
| name | string | Name of web analytic `Visual Website Optimizer` |
```

```JSON title="Sample Request" lineNumbers
GET https://api.bigcommerce.com/stores/{{STORE_HASH}}/v3/settings/analytics/{id}
X-Auth-Token: {{ACCESS_TOKEN}}
Content-Type: application/json
Accept: application/json
```

```JSON title="Sample Response" lineNumbers
{
    "data": {
        "id": 2,
        "channel_id": 0,
        "name": "Visual Website Optimizer",
        "enabled": false,
        "vwo_smartcode": "VWO-1234567890"
    },
    "meta": {}
}
```

### Update the Visual Website Optimizer Analytic

```JSON title="Sample Request" lineNumbers
PUT https://api.bigcommerce.com/stores/{{STORE_HASH}}/v3/settings/analytics/{id}
X-Auth-Token: {{ACCESS_TOKEN}}
Content-Type: application/json
Accept: application/json

{
"id": 2,
"channel_id": 0,
"name": "Visual Website Optimizer",
"enabled": false,
"vwo_smartcode": "VWO-1234567890"
}
```
```JSON title="Sample Response" lineNumbers
{
    "data": {
        "id": 2,
        "channel_id": 0,
        "name": "Visual Website Optimizer",
        "enabled": false,
        "vwo_smartcode": "VWO-1234567890"
    },
    "meta": {}
}
```
&nbsp;

## Facebook Pixel

| Field | Type | Description |
| ------------- | -------- |  -------- |

> | id | integer | Id of the Web Analytic. `1` for Facebook Pixel |

Reviewer 2: ```suggestion
| id | integer | ID of the web analytic. `1` for Facebook Pixel |
```

> | channel_id | integer | Id of the storefront channel. Default is `0`. |

Reviewer 2: ```suggestion
| channel_id | integer | ID of the storefront channel. Default is `0`. |
```

> | name | string | Name of Web Analytic `Facebook Pixel` |

Reviewer 2: ```suggestion
| name | string | Name of web analytic `Facebook Pixel` |
```

| enabled | boolean | Indicates whether merchant has [enabled Facebook Pixel](https://support.bigcommerce.com/s/article/Data-Solutions?language=en_US#web-analytics) in their store |

> | pixel_id | string | Facebook Pixel ID used to connect store to Faceboook Pixel |

Reviewer: ```suggestion
| pixel_id | string | Facebook Pixel ID used to connect store to Facebook Pixel |
```
Unless you want Facebook to be extra spooky

GET https://api.bigcommerce.com/stores/{{STORE_HASH}}/v3/settings/analytics/{id}
X-Auth-Token: {{ACCESS_TOKEN}}
Content-Type: application/json
Accept: application/json
```

```JSON title="Sample Response" lineNumbers
{
    "data": {
        "id": 3,
        "channel_id": 0,
        "name": "Facebook Pixel",
        "enabled": true,
        "pixel_id": "FP-1234567890",
        "is_oauth_connected": true
    },
    "meta": {}
}
```

### Update the Facebook Pixel Analytic

```JSON title="Sample Request" lineNumbers
PUT https://api.bigcommerce.com/stores/{{STORE_HASH}}/v3/settings/analytics/{id}
X-Auth-Token: {{ACCESS_TOKEN}}
Content-Type: application/json
Accept: application/json

{
  "id": 3,
  "channel_id": 0,
  "name": "Facebook Pixel",
  "enabled": true,
  "pixel_id": "FP-1234567890",
  "is_oauth_connected": true
}

```

```JSON title="Sample Response" lineNumbers
{
    "data": {
        "id": 3,
        "channel_id": 0,
        "name": "Facebook Pixel",
        "enabled": true,
        "pixel_id": "FP-1234567890",
        "is_oauth_connected": true
    },
    "meta": {}
}
```
&nbsp;

## Segment

| Field | Type | Description |
| ------------- | -------- |  -------- |

> | id | integer | Id of the Web Analytic. `1` for Segment |

Reviewer 2: ```suggestion
| id | integer | ID of the web analytic. `1` for Segment |
```

> | channel_id | integer | Id of the storefront channel. Default is `0`. |

Reviewer 2: ```suggestion
| channel_id | integer | ID of the storefront channel. Default is `0`. |
```

> | name | string | Name of Web Analytic `Segment` |

Reviewer 2: ```suggestion
| name | string | Name of web analytic `Segment` |
```

| enabled | boolean | Indicates whether merchant has [enabled Segment](https://support.bigcommerce.com/s/article/Data-Solutions?language=en_US#web-analytics) in their store |

> | api_key | string | Segment API key used to connect store to Segment |

Reviewer 2: ```suggestion
| api_key | string | Segment API key used to connect the store to the segment |
```

GET https://api.bigcommerce.com/stores/{{STORE_HASH}}/v3/settings/analytics/{id}
X-Auth-Token: {{ACCESS_TOKEN}}
Content-Type: application/json
Accept: application/json
```

```JSON title="Sample Response" lineNumbers
{
    "data": {
        "id": 4,
        "channel_id": 0,
        "name": "Segment.com",
        "enabled": false,
        "api_key": "SEG-1234567890",
        "data_tag_enabled": true
    },
    "meta": {}
}
```

### Update the Segment Analytic

```JSON title="Sample Request" lineNumbers
PUT https://api.bigcommerce.com/stores/{{STORE_HASH}}/v3/settings/analytics/{id}
X-Auth-Token: {{ACCESS_TOKEN}}
Content-Type: application/json
Accept: application/json

{
  "id": 4,
  "channel_id": 0,
  "name": "Segment.com",
  "enabled": false,
  "api_key": "SEG-1234567890",
  "data_tag_enabled": true
}

```

```JSON title="Sample Response" lineNumbers
{
    "data": {
        "id": 4,
        "channel_id": 0,
        "name": "Segment.com",
        "enabled": false,
        "api_key": "SEG-1234567890",
        "data_tag_enabled": true
    },
    "meta": {}
}
```
&nbsp;

## Site Verification Tags

| Field | Type | Description |
| ------------- | -------- |  -------- |

> | id | integer | Id of the Web Analytic. `1` for Site Verification Tags |

Reviewer 2: ```suggestion
| id | integer | ID of the web analytic. `1` for Site Verification Tags |
```

> | channel_id | integer | Id of the storefront channel. Default is `0`. |

Reviewer 2: ```suggestion
| channel_id | integer | ID of the storefront channel. Default is `0`. |
```

> | name | string | Name of Web Analytic `Site Verification Tags` |

Reviewer 2: ```suggestion
| name | string | Name of web analytic `Site Verification Tags` |
```

        "id": 6,
        "channel_id": 0,
        "name": "Site Verification Tags",
        "enabled": false,
        "verification_tag": "<meta name=\"google-site-verification\" content=\"twLOPnFqwX5JK2-RksTZp_QEkfUPPgfGJ0_UHrEKLcY\" />"
    },
    "meta": {}
}
```

### Update the Site Verification Tags Analytic

```JSON title="Sample Request" lineNumbers
PUT https://api.bigcommerce.com/stores/{{STORE_HASH}}/v3/settings/analytics/{id}
X-Auth-Token: {{ACCESS_TOKEN}}
Content-Type: application/json
Accept: application/json

{
  "id": 6,
  "channel_id": 0,
  "name": "Site Verification Tags",
  "enabled": false,
  "verification_tag": "<meta name=\"google-site-verification\" content=\"twLOPnFqwX5JK2-RksTZp_QEkfUPPgfGJ0_UHrEKLcY\" />"
}

```

```JSON title="Sample Response" lineNumbers
{
    "data": {
        "id": 6,
        "channel_id": 0,
        "name": "Site Verification Tags",
        "enabled": false,
        "verification_tag": "<meta name=\"google-site-verification\" content=\"twLOPnFqwX5JK2-RksTZp_QEkfUPPgfGJ0_UHrEKLcY\" />"
    },
    "meta": {}
}
```

&nbsp;

## Affiliate Conversion Tracking

| Field | Type | Description |
| ------------- | -------- |  -------- |

> | id | integer | Id of the Web Analytic. `1` for Affiliate Conversion Tracking |

Reviewer 2: ```suggestion
| id | integer | ID of the web analytic. `1` for Affiliate Conversion Tracking |
```

> | channel_id | integer | Id of the storefront channel. Default is `0`. |

Reviewer 2: ```suggestion
| channel_id | integer | ID of the storefront channel. Default is `0`. |
```

> | name | string | Name of Web Analytic `Affiliate Conversion Tracking` |

Reviewer 2: ```suggestion
| name | string | Name of web analytic `Affiliate Conversion Tracking` |
```

```JSON title="Sample Request" lineNumbers
GET https://api.bigcommerce.com/stores/{{STORE_HASH}}/v3/settings/analytics/{id}
X-Auth-Token: {{ACCESS_TOKEN}}
Content-Type: application/json
Accept: application/json
```

```JSON title="Sample Response" lineNumbers
{
  "id": 7,
  "channel_id": 0,
  "name": "Affiliate Conversion Tracking",
  "enabled": false,
  "connection": "<script>js code here...</script>"
}
```

### Update the Affiliate Conversion Tracking Analytic

```JSON title="Sample Request" lineNumbers
PUT https://api.bigcommerce.com/stores/{{STORE_HASH}}/v3/settings/analytics/{id}
X-Auth-Token: {{ACCESS_TOKEN}}
Content-Type: application/json
Accept: application/json

{
    "data": {
        "id": 7,
        "channel_id": 0,
        "name": "Affiliate Conversion Tracking",
        "enabled": false,
        "connection": "<script>js code here...</script>"
    },
    "meta": {}
}

```

```JSON title="Sample Response" lineNumbers
{
    "data": {
        "id": 7,
        "channel_id": 0,
        "name": "Affiliate Conversion Tracking",
        "enabled": false,
        "connection": "<script>js code here...</script>"
    },
    "meta": {}
}
```

&nbsp;

> ## Get All Web Analytics

Reviewer 2: ```suggestion
## Get all web analytics
```

> All six web analytics will be returned. As shown, fields for codes (e.g. `verification_tag`) in which a merchant has not entered a value will return as an empty string.     

Reviewer: Recommended to use "such as" or "for example" instead of "e.g." https://developers.google.com/style/word-list#eg

            "enabled": false,
            "vwo_smartcode": "VWO-1234567890"
        },
        {
            "id": 3,
            "channel_id": 0,
            "name": "Facebook Pixel",
            "enabled": true,
            "pixel_id": "FP-1234567890",
            "is_oauth_connected": true
        },
        {
            "id": 4,
            "channel_id": 0,
            "name": "Segment.com",
            "enabled": false,
            "api_key": "SEG-1234567890",
            "data_tag_enabled": true
        },
        {
            "id": 6,
            "channel_id": 0,
            "name": "Site Verification Tags",
            "enabled": false,
            "verification_tag": ""
        },
        {
            "id": 7,
            "channel_id": 0,
            "name": "Affiliate Conversion Tracking",
            "enabled": false,
            "connection": "<script>js code here...</script>"
        }
    ],
    "meta": {}
}
```

&nbsp;

## Related Resources
- [Overview Web Analytics](https://support.bigcommerce.com/s/article/Data-Solutions?language=en_US#web-analytics)
- [Google Analytics](https://support.bigcommerce.com/s/article/Setting-Up-Google-Analytics?language=en_US)

> - [Visual Website Optimizer](...)

Reviewer: I assume `(...)` is a placeholder, but a reminder to add a URL

Author: It turns out we don't have a KB doc, so I'll remove the placeholder

- [Facebook Pixel](https://support.bigcommerce.com/s/article/Facebook-Pixel?language=en_US)
- [Segment](https://support.bigcommerce.com/s/article/Setting-up-Segment-com?language=en_US)

> - [Site Verification Tags](...)

Reviewer: I assume `(...)` is a placeholder, but a reminder to add a URL

Author: It turns out we don't have a KB doc, so I'll remove the placeholder

# Data Solutions API

The Data Solutions API lets you configure **storefront** channel settings for a store's prebuilt data analytic solutions.  

Note that a store has global settings for data solutions, from which any storefront channel can inherit. A merchant can override these global settings with storefront-specific settings for their data solutions. This API configures settings for **storefront** channels at the store's global level (`channel_id` is `0`) and at the specific storefront channel level (by specifying a `channel_id` in the query). If no channel ID is specified in the query, the default channel is the global channel `0`.    

<!-- theme: info -->

> > **Note**: You can obtain storefront channel IDs using [Get All Channels](/api-reference/store-management/channels/channels/listchannels). 

Reviewer 3: ```suggestion
> #### Note
> You can obtain storefront channel IDs using the [Get all channels](/api-reference/store-management/channels/channels/listchannels) endpoint. 
```
